File: ghidra_project/disasm_ref.py
# Ghidra Python script to disassemble refs to a specific address
from ghidra.program.model.address import Address
from ghidra.program.model.listing import Instruction

def run():
    # Sony MDCT window table address
    table_addr_val = 0x452340
    table_addr = currentProgram.getAddressFactory().getAddress(hex(table_addr_val))
    
    print("Searching for references to {}".format(table_addr))
    
    refs = getReferencesTo(table_addr)
    for ref in refs:
        ref_addr = ref.getFromAddress()
        print("Reference found at: {}".format(ref_addr))
        
        instr = getInstructionAt(ref_addr)
        if instr:
            print("Instruction: {}".format(instr))
            # Print 30 instructions to capture the loop
            curr = instr.getNext()
            for i in range(30):
                if not curr:
                    break
                print("  {}: {}".format(curr.getAddress(), curr))
                curr = curr.getNext()

if __name__ == "__main__":
    run()

File: ghidra_project/DisasmRef.java
import ghidra.app.script.GhidraScript;
import ghidra.program.model.address.Address;
import ghidra.program.model.listing.Instruction;
import ghidra.program.model.listing.InstructionIterator;

class DisasmRef extends GhidraScript {
    @Override
    public void run() throws Exception {
        // Offset found was 0x178FD. 
        // Image base is likely 0x400000. 
        // PE headers usually 0x400... let's try to find it by file offset logic if possible.
        // Or just use the address we searched: 0x452340.
        
        Address tableAddr = toAddr(0x452340);
        println("Searching for references to " + tableAddr);
        
        for (Address ref : getReferencesTo(tableAddr)) {
            println("Reference found at: " + ref);
            Instruction instr = getInstructionAt(ref);
            if (instr != null) {
                println("Instruction: " + instr.toString());
                // Print a few more instructions after
                Instruction next = instr.getNext();
                for (int i = 0; i < 20 && next != null; i++) {
                    println("  " + next.getAddress() + ": " + next.toString());
                    next = next.getNext();
                }
            }
        }
    }
}
